wp-content/plugins/tw-contributors/tw-contributors.php

```diff
@@ -56,3 +56,204 @@ function tw_contributors_taxonomy() {
 	register_taxonomy( 'contributor', array( 'post', 'attachment', 'segment' ), $args );
 }
 add_action( 'init', 'tw_contributors_taxonomy', 0 );
+
+/**
+ * Change RSS feed author name.
+ *
+ * @param string $author
+ *
+ * @return string
+ */
+function tw_contributors_rss_author( $author ) {
+
+	// Check if we're in a feed.
+	if ( is_feed() ) {
+
+		$post_id   = get_the_ID();
+		$post_type = get_post_type( $post_id );
+
+		// List of affected post types.
+		$affected_post_types = array( 'post', 'episode', 'segment' );
+
+		// Bail early if not an affected post type.
+		if ( ! in_array( $post_type, $affected_post_types, true ) ) {
+			return $author;
+		}
+
+		$contributors = tw_contributors_get_post_contributors( $post_id );
+
+		// Set first contributor as author.
+		$author = reset( $contributors );
+
+		// If more than one contributor, add action to rss2 action.
+		if ( count( $contributors ) > 1 ) {
+			add_action( 'rss2_item', 'tw_contributors_rss_extra_contributors' );
```

Comment:
Do we need to conditionally add this action hook? Is `the_author` too late for a `rss2_item` action to be registered? The `rss2_item` callback contains logic to make it work regardless of the conditional in `the_author` callback, so why not just always register it. If the concern is to not run `tw_contributors_get_post_contributors` twice, that function should be leveraging caching or transients anyway (if it is slow).

Reply:
The author filter will run before the `rss2_item` and if necessary it will add the callback.
You are right, the idea was trying to avoid running`tw_contributors_get_post_contributors` unnecessarily just using a count method. 
We can remove that conditional and run it always because the query should be cached, so it is more up to what we prefer. 
We are ok removing it if you think it will be more clear.